e2e/tests/upgrades/genesis_test.go

```diff
@@ -0,0 +1,159 @@
+package upgrades
+
+import (
+	"context"
+	"os"
+	"time"
+	"encoding/json"
+	"testing"
+
+	"github.com/cosmos/ibc-go/e2e/testconfig"
+	"github.com/cosmos/ibc-go/e2e/testsuite"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+	cosmos "github.com/strangelove-ventures/interchaintest/v7/chain/cosmos"
+	test "github.com/strangelove-ventures/interchaintest/v7/testutil"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/ibc-go/e2e/testvalues"
+)
+type GenesisState map[string]json.RawMessage
+
+func TestGenesisTestSuite(t *testing.T) {
+	suite.Run(t, new(GenesisTestSuite))
+}
+
+type GenesisTestSuite struct {
+	testsuite.E2ETestSuite
+	cosmos.ChainNode
+}
+
+func (s *GenesisTestSuite) TestIBCGenesis() {
+	t := s.T()
+
+	configFileOverrides := make(map[string]any)
+	appTomlOverrides := make(test.Toml)
+	configFileOverrides["halt-height"] = haltHeight
+
+	testconfig.DefaultChainOptions().ChainAConfig.ConfigFileOverrides = appTomlOverrides
+
+	ctx := context.Background()
+	relayer, channelA := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
```

Comment:
I believe this isn't working as intended since the chain config instance returned by a call to `testconfig.DefaultChainOptions()` will be different than the instance later called by `SetupChainsRelayerAndChannel`. The fix is fairly simple though:
```suggestion
	configFileOverrides := make(map[string]any)
	appTomlOverrides := make(test.Toml)
	
	appTomlOverrides["halt-height"] = haltHeight
	configFileOverrides["config/app.toml"] = appTomlOverrides

	chainOpts := func(options *ChainOptions) {
	    options.ChainAConfig.ConfigFileOverrides = configFileOverrides
	}
	
	// create chains with specified chain configuration options
	chainA, chainB := s.GetChains(chainOpts)
	
	ctx := context.Background()
	relayer, channelA := s.SetupChainsRelayerAndChannel(ctx)
```

I would recommend committing these changes locally as I haven't tried to format/compile